cli_tools/common/disk/inspect.go

```diff
@@ -63,50 +71,197 @@ func NewInspector(wfAttributes daisycommon.WorkflowAttributes, network string, s
 	daisycommon.SetWorkflowAttributes(wf, wfAttributes)
 	wf.Vars["network"] = daisy.Var{Value: network}
 	wf.Vars["subnet"] = daisy.Var{Value: subnet}
-	return &defaultInspector{wf}, nil
+	return &bootInspector{[]string{}, &defaultDaisyWorker{wf}}, nil
 }
 
-// defaultInspector implements disk.Inspector using a Daisy workflow.
-type defaultInspector struct {
-	wf *daisy.Workflow
+// bootInspect implements disk.Inspector using the Python boot-inspect package,
+// executed on a worker VM using Daisy.
+type bootInspector struct {
+	traceLogs []string
+	worker    daisyWorker
+}
+
+// daisyWorker is a facade over daisy.Workflow to facilitate mocking.
+type daisyWorker interface {
+	runAndReadSerialValue(key string, vars map[string]string) (string, error)
+	cancel(reason string) bool
+	traceLogs() []string
+}
+
+func (i *bootInspector) Cancel(reason string) bool {
+	i.tracef("Canceling with reason: %q", reason)
+	return i.worker.cancel(reason)
+}
+
+func (i *bootInspector) TraceLogs() []string {
+	var combined []string
+	combined = append(combined, i.traceLogs...)
+	combined = append(combined, i.worker.traceLogs()...)
+	return combined
 }
 
 // Inspect finds partition and boot-related properties for a GCP persistent disk, and
 // returns an InspectionResult. `reference` is a fully-qualified PD URI, such as
 // "projects/project-name/zones/us-central1-a/disks/disk-name". `inspectOS` is a flag
 // to determine whether to inspect OS on the disk.
-func (inspector *defaultInspector) Inspect(reference string, inspectOS bool) (ir InspectionResult, err error) {
-	inspector.wf.AddVar("pd_uri", reference)
-	inspector.wf.AddVar("is_inspect_os", strconv.FormatBool(inspectOS))
-	err = inspector.wf.Run(context.Background())
+func (i *bootInspector) Inspect(reference string, inspectOS bool) (InspectionResult, error) {
```

Comment:
Related to the comment about renaming bootInspector to maybe bootDiskInspector, looking at this func signature, it's not clear what is being inspected, since neither the func, nor reference arg mention disk. Both could have "disk" in their names to improve readability.

Reply:
Similar to above, I don't have a huge preference. That said, I took this approach since the symbol would be `disk.Inspector` when consumed outside of this package.